nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/QueryCassandra.java

```diff
@@ -75,19 +81,64 @@
+        + "may be converted to Avro, JSON or CSV format. Streaming is used so arbitrarily large result sets are supported. This processor can be "
         + "scheduled to run on a timer, or cron expression, using the standard scheduling methods, or it can be triggered by an incoming FlowFile. "
         + "If it is triggered by an incoming FlowFile, then attributes of that FlowFile will be available when evaluating the "
         + "select query. FlowFile attribute 'executecql.row.count' indicates how many rows were selected.")
+@Stateful(scopes = Scope.CLUSTER, description = "After performing query, the maximum value of the specified column is stored, "
+        + "fetch all rows whose values in the specified Maximum Value column(s) are larger than the previously-seen maximum"
+        + "State is stored across the cluster so that the next time this Processor can be run with min and max values")
 @WritesAttributes({@WritesAttribute(attribute = "executecql.row.count", description = "The number of rows returned by the CQL query")})
 public class QueryCassandra extends AbstractCassandraProcessor {
 
+    public static final String CSV_FORMAT = "CSV";
     public static final String AVRO_FORMAT = "Avro";
     public static final String JSON_FORMAT = "JSON";
 
+    public static final String CASSANDRA_WATERMARK_MIN_VALUE_ID = "CASSANDRA_WATERMARK_MIN_VALUE_ID";
+    public static final String CASSANDRA_WATERMARK_MAX_VALUE_ID = "CASSANDRA_WATERMARK_MAX_VALUE_ID";
+
     public static final String RESULT_ROW_COUNT = "executecql.row.count";
 
+    public static final PropertyDescriptor INIT_WATERMARK = new PropertyDescriptor.Builder().name("Initial Watermark Value")
+            .description("Use it only once.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor BACKOFF_PERIOD = new PropertyDescriptor.Builder()
+            .name("Backoff Period")
+            .description("Only records older than the backoff period will be eligible for pickup. This can be used in the ILM use case to define a retention period.")
+            .defaultValue("10 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor OVERLAP_TIME = new PropertyDescriptor.Builder()
+            .name("Overlap Period")
+            .description("Amount of time to overlap into the last load date to ensure long running transactions missed by previous load weren't missed. Recommended: >0s")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("0 seconds")
+            .build();
+
+    public static final PropertyDescriptor DATE_FIELD = new PropertyDescriptor.Builder()
```

Comment:
I think it's a good idea, but a little bit complex. 
because we usually want to concentrate the time of record when we use the incremental fetch feature. ( To avoid omissions )

So there are the concepts of BACKOFF_PERIOD and OVERLAP_TIME.
It is difficult to apply the other type column.

In my opinion, a date column is most appropriate for  the incremental fetch feature.